osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcess.java

```diff
@@ -51,11 +56,12 @@
 
     private String ipAddress;
 
-    public RecoverKeyProcess(final DomainHelperService domainHelperService,
-            final DlmsDeviceRepository dlmsDeviceRepository, final int responseTimeout, final int logicalDeviceAddress,
-            final DlmsDeviceAssociation deviceAssociation) {
+    @Autowired
+    private SecretManagementService secretManagementService;
```

Comment:
Is there any rationale behind auto-wiring the `secretManagementService` as a field when the other dependencies are set through the constructor?

Reply:
No intentional rationale. But after inspecting it, it seems correct te me: the other dependencies are set in DlmsConfig and I think that is the wrong place to wire it. So looks okay to me.